mmcv/cnn/rfsearch/search.py

```diff
@@ -0,0 +1,249 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import logging
+import os
+from typing import Dict
+
+import torch  # noqa
+import torch.nn as nn
+
+from mmcv.runner import HOOKS, Hook
+from .operator import Conv2dRFSearchOp, ConvRFSearchOp  # noqa
+from .utils import load_structure, write_to_json
+
+logging.basicConfig(
+    format='[%(asctime)s-%(filename)s-%(levelname)s:%(message)s]',
+    level=logging.NOTSET,
+    datefmt='%Y-%m-%d%I:%M:%S %p',
+)
+logger = logging.getLogger('Searcher')
+logger.setLevel(logging.ERROR)
+
+
+@HOOKS.register_module()
+class RFSearch(Hook):
+    """Rcecptive field search via dilation rates.
+        Paper: RF-Next: Efficient Receptive Field
+            Search for Convolutional Neural Networks
+
+    Args:
+        logdir (str, optional): save path of searched structure.
+                        Defaults to './log'.
+        mode (str, optional):
+                search/fixed_single_branch/fixed_multi_branch.
+        config (Dict, optional): config dict of search.
+        rfstructure_file (Optional[str], optional):
+                searched recptive fields of the model.
+    """
+
+    def __init__(self,
+                 logdir: str = './log',
+                 mode: str = 'search',
```

Comment:
It's better to use the work_dir as logdir.

Reply:
Any suggestions to get the work_dir in the hook? It seems the cfg.work_dir in mmdet is defined in the train.py file.

Reply:
`work_dir` will be attached at `runner`. We can directly use `runner.work_dir` when we want to save some information.